setup.py

```diff
@@ -34,18 +48,49 @@ def build_tables():
            outputdir='xonsh')
     sys.path.pop(0)
 
+
+def install_jupyter_hook():
+    if not HAVE_JUPYTER:
+        print('Could not install Jupyter kernel spec, please install Jupyter/IPython.')
+        return
+    spec = {"argv": [sys.executable, "-m", "xonsh.jupyter_kernel", 
+                                     "-f", "{connection_file}"],
+            "display_name":"Xonsh",
+            "language":"xonsh",
+            "codemirror_mode":"shell",
+            }
+    if CONDA:
+        d = os.path.join(os.environ['PREFIX'] +
```

Comment:
Yes. You are right. It has been a while since I tingled with these things. It is of course python in the build environment that runs setup.py, so it should be the same sys.prefix, when building. 

I am not too happy about this being in the setup.py file, but I haven't seen any good ways of installing the kernel_spec with conda. You did something for the r-kernel, right? I think I saw it on conda-recipes, but I can't seem to find it now. May it should be a post-install script that tests if Jupyter is there, and then runs the install_kernel_spec() function.